### PR TITLE
*: forbid set tiflash replica count for a placement table

### LINKED ISSUE #30741: Forbid tiflash replica on the table that has placement settings

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #30741

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
alter table/partition placement will fail for the table with tiflash replica 
set tiflash replica will fail for the table with placement options
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

**Comment by bb7133** [comment]
Please fix the lint check: https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check/detail/tidb_ghpr_check/40089/pipeline/

File: ddl/placement_sql_test.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ import (
 	"fmt"
 	"sort"
 
+	"github.com/pingcap/tidb/ddl"
+
```
Comment (AilinKid) [inline ddl/placement_sql_test.go:23]: remove the blank line

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

**Review by AilinKid — APPROVED** [review]

**Pull request merged**